msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/ManagedIdentityApplication.java

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4j;
+
+import com.nimbusds.oauth2.sdk.auth.ClientAuthentication;
+import lombok.Getter;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Class to be used to acquire tokens for managed identity.
+ * For details see {@link IManagedIdentityApplication}
+ * <p>
+ * Conditionally thread-safe
+ */
+public class ManagedIdentityApplication extends AbstractClientApplicationBase implements IManagedIdentityApplication {
+
+    private String resource;
+
+    @Getter
+    private ManagedIdentityId managedIdentityId;
+
+    private ManagedIdentityApplication(Builder builder) {
+        super(builder);
+        this.managedIdentityId = builder.managedIdentityId;
+        log = LoggerFactory.getLogger(ManagedIdentityApplication.class);
+    }
+
+    /**
+     * Creates instance of Builder of ManagedIdentityApplication
+     *
+     * @param managedIdentityId ManagedIdentityId to specify if it System Assigned or User Assigned
+     *                          and provide id if it is user assigned.
+     * @return instance of Builder of ManagedIdentityApplication
+     */
+    public static Builder builder(ManagedIdentityId managedIdentityId) {
+        return new Builder(managedIdentityId);
+    }
+
+    @Override
+    public CompletableFuture<IAuthenticationResult> acquireTokenForManagedIdentity(ManagedIdentityParameters managedIdentityParameters)
+            throws Exception {
+        RequestContext requestContext = new RequestContext(
+                this,
+                managedIdentityId.getIdType() == ManagedIdentityIdType.SystemAssigned ?
+                        PublicApi.ACQUIRE_TOKEN_BY_SYSTEM_ASSIGNED_MANAGED_IDENTITY :
+                        PublicApi.ACQUIRE_TOKEN_BY_USER_ASSIGNED_MANAGED_IDENTITY,
+                managedIdentityParameters);
+
+        ManagedIdentityRequest managedIdentityRequest = new ManagedIdentityRequest(this, requestContext);
+
+        return this.executeRequest(managedIdentityRequest);
+    }
+
+    @Override
+    protected ClientAuthentication clientAuthentication() {
```

Comment:
What does this do? protected means it's part of the public api

Reply:
This is further in the flow needed to build the OAuth request and I think not needed for managed identity. https://github.com/AzureAD/microsoft-authentication-library-for-java/blob/6106d521e6760bd5cf17a7b32f410e9e125fd4f1/msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/TokenRequestExecutor.java#L81